### PR TITLE
feat: Password reset page

### PR DESCRIPTION
**Pull request by samuliys**
## WHY:

Advances #189 

### What has been changed (if possible, add screenshots, gifs, etc. )

Implements the password reset page, which is used to set a new password.
The new password is validated against the same password rules as in registration.

To test the feature send the following request
`curl http://127.0.0.1:1337/api/auth/forgot-password -H "Content-Type: application/json" -d '{"email":"first.last@example.com"}'`

Then, navigate to http://localhost:1080/ and from the email copy the code and navigate to http://localhost:5173/candidate/password-reset?code=??? where ??? is the code you received. Then, after successfully changing the password, you are redirected to /candidate where you can log in using the new password (email is [first.last@example.com](mailto:first.last@example.com))`

## Check off each of the following tasks as they are completed

- [x] I have reviewed the changes myself in this PR. Please check the [self-review document](https://github.com/OpenVAA/voting-advice-application/blob/main/docs/contributing/self-review.md)
- [ ] I have added or edited unit tests.
- [ ] I have run the unit tests successfully.
- [x] I have tested this change on my own device.
- [ ] I have tested this change on other devices (Using Browserstack is recommended).
- [x] I have tested my changes using the [WAVE extension](https://wave.webaim.org/extension/)
- [x] I have added documentation where necessary.
- [x] Is there an existing issue linked to this PR?

**Clean up your git commit history before submitting the pull request!**

### CODE REVIEW

File: frontend/src/routes/candidate/password-reset/PasswordResetPage.svelte [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+      required />
+    {#if errorMessage}
+      <p class="text-center text-error">
+        {errorMessage}
+      </p>
+    {/if}
+
+    <button class="btn btn-primary mb-md w-full" type="submit" disabled={!disableSetButton}>
+      {$_('candidateApp.setPassword.setPassword')}
+    </button>
```
Comment (anolpe) [inline frontend/src/routes/candidate/password-reset/PasswordResetPage.svelte:106]: This is not using the new button component.

File: frontend/src/routes/candidate/password-reset/PasswordResetPage.svelte [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  export let code: string;
+
+  let password = '';
+  let passwordConfirmation = '';
+  let validPassword = false;
+  let errorMessage = '';
+
+  $: disableSetButton = validPassword && passwordConfirmation.length > 0;
+
```
Comment (anolpe) [inline frontend/src/routes/candidate/password-reset/PasswordResetPage.svelte:22]: Should this be named differently? Now it is `false` when the button should be disabled.

**Review by anolpe — CHANGES_REQUESTED** [review]
I didn't review Stepan's commit because that will be in another PR(?)

**Review by suviz — CHANGES_REQUESTED** [review]
Fix Antti's suggestions. Otherwise looks good!

**Review by anolpe — APPROVED** [review]
LGTM!

**Review by suviz — APPROVED** [review]
LGTM!

**Pull request merged**